specs/capella/light-client/full-node.md

```diff
@@ -76,3 +76,31 @@ def block_to_light_client_header(block: SignedBeaconBlock) -> LightClientHeader:
         execution_branch=execution_branch,
     )
 ```
+
+## Deriving light client data
+
+### `create_light_client_update`
+
+Full nodes SHOULD continue deriving the `LightClientUpdate` for each sync committee period as per the altair specs but with the modified or upgraded `attested_header` and `finalized_header` with the following considerations:
+
+- Both `attested_header` and `finalized_header` are of the same fork types as `LightClientUpdate` and might need upgradation to bundle as update.
+- Above implies that the first update of capella MIGHT have both `attested_header` and `finalized_header` upgraded (if capella hardfork coincides with a new period) as they might belong to pre-capella epochs. Post this `finalized_header` will need upgradation till a new finalization occurs in capella.
```

Comment:
The `attested_header` determines the spec fork to use as basis for building the data.
This is mentioned in the Altair [p2p-interface.md](https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/p2p-interface.md).

> The gossip ForkDigest-context is determined based on `compute_fork_version(compute_epoch_at_slot(optimistic_update.attested_header.beacon.slot))`.

The entire object will be built using the logic from that spec fork. 

The first case cannot really happen for data production, the `attested_header` will never need an upgrade, as it is the one defining the spec fork. Spec fork is not determined by wall clock or `signature_slot`.

Also, the "first update" is not the only one that's special, depending on branching situation and missed blocks, you could even set up hypothetical scenarios where `finalized` is altair, `attested` is capella, and `signature_slot` is deneb. the unit tests construct some of those cases to cover edge cases